### PR TITLE
test(RBAC): backend integration tests for RBAC for releases

### PR DESCRIPTION
**Pull request by kjaskiewiczz**
Ticket: [MEN-7074](https://northerntech.atlassian.net/browse/MEN-7074)

[MEN-7074]: https://northerntech.atlassian.net/browse/MEN-7074?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by kjaskiewiczz** [comment]
@mender-test-bot start pipeline --fast --pr deployments-enterprise/1034

**Comment by kjaskiewiczz** [comment]
@mender-test-bot start pipeline --fast --pr deployments-enterprise/1034

**Comment by kjaskiewiczz** [comment]
@mender-test-bot start pipeline --fast --pr deployments-enterprise/1034

File: backend-tests/tests/test_rbac_v2.py [hunk 1]
```diff
@@ -1508,3 +1511,601 @@ def test_get_emails_by_group(self, clean_mongo, test_case):
...
+            tenant.users[0].token, test_case["roles"], status_code=201,
+        )
+        # create and upload artifacts
+        artifacts = (
+            {"artifact_name": "foo", "device_types": ["arm1"], "size": 256},
+            {"artifact_name": "bar", "device_types": ["arm1"], "size": 256},
+        )
+        api_client = ApiClient(deployments.URL_MGMT)
+        api_client_v2 = ApiClient(deployments_v2.URL_MGMT)
+        api_client.headers = {}  # avoid default Content-Type: application/json
```
Comment (merlin-northern) [inline backend-tests/tests/test_rbac_v2.py:1800]: ok so without that the default will be json Content-type, right?

File: backend-tests/tests/test_rbac_v2.py [hunk 2]
```diff
@@ -1508,3 +1511,601 @@ def test_get_emails_by_group(self, clean_mongo, test_case):
...
+            assert r.status_code == 201
+
+        # get all releases
+        api_client_v2.with_auth(tenant.users[0].token)
+        r = api_client_v2.call("GET", deployments_v2.URL_RELEASES)
+        assert r.status_code == 200
+        all_releases = r.json()
+        assert len(all_releases) == 2
+
+        # switch to user with limited access
```
Comment (merlin-northern) [inline backend-tests/tests/test_rbac_v2.py:1829]: ok just for me, and to double check which error code represents the limits?

File: backend-tests/tests/test_rbac_v2.py [hunk 3]
```diff
@@ -1508,3 +1511,601 @@ def test_get_emails_by_group(self, clean_mongo, test_case):
...
+        all_artifacts = r.json()
+        assert len(all_artifacts) == 2
+        foo_artifact_id = ""
+        bar_artifact_id = ""
+        for a in all_artifacts:
+            if a["name"] == "bar":
+                bar_artifact_id = a["id"]
+            elif a["name"] == "foo":
+                foo_artifact_id = a["id"]
+
```
Comment (merlin-northern) [inline backend-tests/tests/test_rbac_v2.py:1998]: shall we add assert that we got both ids?

File: backend-tests/tests/test_rbac_v2.py [hunk 4]
```diff
@@ -1508,3 +1511,601 @@ def test_get_emails_by_group(self, clean_mongo, test_case):
...
+                files=(
+                    ("description", (None, "description")),
+                    ("size", (None, str(os.path.getsize(artifact)))),
+                    (
+                        "artifact",
+                        (artifact, open(artifact, "rb"), "application/octet-stream",),
+                    ),
+                ),
+            )
+        assert r.status_code == 403
```
Comment (merlin-northern) [inline backend-tests/tests/test_rbac_v2.py:2043]:
```suggestion
            assert r.status_code == 403
```
probably does not make any difference

**Review by merlin-northern — APPROVED** [review]
huge amount of work. gooks good.

File: backend-tests/tests/test_rbac_v2.py [hunk 5]
```diff
@@ -1508,3 +1511,601 @@ def test_get_emails_by_group(self, clean_mongo, test_case):
...
+
+        # create deployment
+        # will be fixed with MEN-7165
+        # request_body = {
+        #    "name": "dummy-deployment",
+        #    "artifact_name": "bar",
+        #    "devices": ["1"],
+        # }
+        # resp = api_client.call("POST", "/deployments", body=request_body)
+        # assert resp.status_code == 403
```
[the remark this answers is not in the document]
Comment (kjaskiewiczz) [reply backend-tests/tests/test_rbac_v2.py:1893]: This one will fail until MEN-7165 is done (as comment suggests)

File: backend-tests/tests/test_rbac_v2.py (hunk 1 above)
Comment (kjaskiewiczz) [reply backend-tests/tests/test_rbac_v2.py:1800]: yes; and for uploading artifacts we need different one

File: backend-tests/tests/test_rbac_v2.py (hunk 2 above)
Comment (kjaskiewiczz) [reply backend-tests/tests/test_rbac_v2.py:1829]: not sure I understand the question

**Pull request merged**